sway-lib-std/src/contract_id.sw

```diff
@@ -233,6 +233,33 @@ impl AssetId {
 }
 
 impl ContractId {
+    /// Returns the ContractId of the currently executing contract.
+    ///
+    /// # Additional Information
+    ///
+    /// This is equivalent to std::callframes::contract_id().
+    ///
+    /// **_Note:_** If called in an external context, this will **not** return a ContractId.
+    /// If called externally, will actually return a pointer to the Transaction Id (Wrapped in the ContractId struct).
+    /// 
+    /// # Returns
+    ///
+    /// * [ContractId] - The contract id of this contract.
+    ///
+    /// # Examples
+    ///
+    /// ```sway
+    /// use std::{constants::ZERO_B256, token::mint};
+    ///
+    /// fn foo() {
+    ///     let this_contract = ContractId::this();
+    ///     mint(ZERO_B256, 50);
+    ///     Address::from(ZERO_B256).transfer(AssetId::default(this_contract), 50);
+    /// }
+    /// ```
+    pub fn this() -> ContractId {
```

Comment:
```suggestion
    pub fn id() -> ContractId {
```

Let's not introduce that Js / Ts nonsense into this

Reply:
We already kinda have with this_balance

Reply:
That's because we also have `balance_of` iirc or another function.

Either way, "this" makes absolutely no sense when you read the name alone. You need to have knowledge to know how to implicitly use it. "id" makes more sense.

Reply:
I'd lean towards "this", because `ContractId::id()` doesn't imply what you're getting the ID of

Reply:
Yes it does, you're calling it on the `ContractId` type. Why would it get the id of another contract?

Reply:
Yeah i agree with david, just `id` as the name of a function tells me nothing about what its doing

Reply:
neither that the word "this", "that", "applepie" etc.